### PR TITLE
Integration test for sys tables

### PR DESCRIPTION
**Pull request by surekhasaharan**
Add integration test for sys tables

### CODE REVIEW

File: integration-tests/src/test/resources/queries/sys_segment_queries.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        "server_type": "historical"
+      }
+    ]
+  },
+  {
+    "sqlQuery": {
+      "query": "SELECT * FROM sys.tasks where status='FAILED'"
+    },
+    "expectedResults": [
+    ]
```
Comment (nishantmonu51) [inline integration-tests/src/test/resources/queries/sys_segment_queries.json]: empty result is not much helpful for validation. 
It wold be better if we can submit a task and then query for the status of it.

Comment (surekhasaharan) [reply integration-tests/src/test/resources/queries/sys_segment_queries.json]: sounds good, I added another test to query sys table for batch index tasks

File: integration-tests/src/main/java/org/apache/druid/testing/utils/SqlQueryWithResults.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import java.util.Map;
+
+public class SqlQueryWithResults
+{
+  private final SqlQuery sqlQuery;
+  private final List<Map<String, Object>> expectedResults;
+
+
+  @JsonCreator
+  @JsonIgnoreProperties(ignoreUnknown = true)
```
Comment (jihoonson) [inline integration-tests/src/main/java/org/apache/druid/testing/utils/SqlQueryWithResults.java]: What properties can be unknown and ignored?

**Comment by surekhasaharan** [comment]
> It looks that many codes are duplicated in TestQueryHelper and SqlResourceTestClient. Could you export them as a common abstract class or a util class?
@jihoonson You probably meant QueryResourceTestClient and SqlResourceTestClient, I added an abstract class for those.

File: integration-tests/src/main/java/org/apache/druid/testing/utils/SqlQueryWithResults.java (hunk 2 above)
Comment (surekhasaharan) [reply integration-tests/src/main/java/org/apache/druid/testing/utils/SqlQueryWithResults.java]: I think I was trying out something, this should not be required anymore.

File: integration-tests/src/main/java/org/apache/druid/testing/utils/SqlQueryWithResults.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  public SqlQueryWithResults(
+      @JsonProperty("sqlQuery") SqlQuery query,
+      @JsonProperty("expectedResults") List<Map<String, Object>> expectedResults
+  )
+  {
+    this.sqlQuery = query;
+    this.expectedResults = expectedResults;
+  }
+
```
[the remark this answers is not in the document]
Comment (surekhasaharan) [reply integration-tests/src/main/java/org/apache/druid/testing/utils/SqlQueryWithResults.java]: done

File: integration-tests/src/main/java/org/apache/druid/testing/utils/SqlQueryWithResults.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import org.apache.druid.sql.http.SqlQuery;
+
+import java.util.List;
+import java.util.Map;
+
+public class SqlQueryWithResults
+{
+  private final SqlQuery sqlQuery;
+  private final List<Map<String, Object>> expectedResults;
+
```
[the remark this answers is not in the document]
Comment (surekhasaharan) [reply integration-tests/src/main/java/org/apache/druid/testing/utils/SqlQueryWithResults.java]: done

**Comment by jihoonson** [comment]
@surekhasaharan so sorry, I wasn't clear. I meant there're also some duplications in `TestQueryHelper` like `testSqlQueriesFromFile` and `testQueriesFromFile`. Can we remove them too?

**Comment by surekhasaharan** [comment]
@jihoonson oops, I misunderstood your comment. I have added `AbstractTestQueryHelper` to keep common code for `TestQueryHelper` and `SqlTestQueryHelper`

**Comment by jihoonson** [comment]
@surekhasaharan thanks. It loos good to me. Would you resolve the conflicts?

**Pull request merged**